base_content_space.sol

```diff
@@ -294,9 +294,13 @@ contract BaseFactory is Ownable {
 
     bytes32 public version ="BaseFactory20200316120700ML"; //class name (max 16), date YYYYMMDD, time HHMMSS and Developer initials XX
 
+    // TODO: similar issue as tx.origin in Ownable - can't use msg.sender here because it's the space. But as with ownable,
+    //  don't think there's a legit spoofing attack here because the spoofee ends up with the rights.
     function createContentType() public returns (address) {
         address newType = (new BaseContentType(msg.sender));
-        BaseContentType(newType).setRights(tx.origin, 0 /*TYPE_SEE*/, 2 /*ACCESS_CONFIRMED*/); // register library in user wallet
+        BaseContentType theType = BaseContentType(newType);
+        theType.setRights(tx.origin, 0 /*TYPE_SEE*/, 2 /*ACCESS_CONFIRMED*/); // register library in user wallet
+        theType.transferOwnership(tx.origin);
         return newType;
     }
 
```

Comment:
In base_content_space.sol, we can modify the BaseContentSpace implementation of createContent:

 function createContent(address lib, address content_type) public returns (address) {
        require(msg.sender == lib);
        address contentAddress = BaseContentFactory(contentFactory).createContent(lib, content_type);
        emit CreateContent(contentAddress);
        return contentAddress;
    }


This way, we have ensured that the end user called the function on the lib, that the lib called the function on the content space and that the content space called the method on the factory, the the tx.origin is guaranteed to be the original end user.

Reply:
Done